fe/fe-core/src/main/java/com/starrocks/consistency/MetaRecoveryDaemon.java

```diff
@@ -0,0 +1,347 @@
+                    LOG.warn("recover partition version failed, table is null, versionInfo: {}", version);
+                    continue;
+                }
+                if (!table.isOlapTableOrMaterializedView()) {
+                    LOG.warn("recover partition version failed, table is not OLAP table, versionInfo: {}", version);
+                    continue;
+                }
+                PhysicalPartition physicalPartition = table.getPhysicalPartition(version.getPartitionId());
+                if (physicalPartition == null) {
+                    LOG.warn("recover partition version failed, partition is null, versionInfo: {}", version);
+                    continue;
+                }
+
+                Partition partition = table.getPartition(physicalPartition.getParentId());
+                if (partition == null) {
+                    LOG.warn("recover partition version failed, partition is null, versionInfo: {}", version);
+                    continue;
+                }
+
+                long originVisibleVersion = physicalPartition.getVisibleVersion();
+                long originNextVersion = physicalPartition.getNextVersion();
+                physicalPartition.setVisibleVersion(version.getVersion(), recoveryInfo.getRecoverTime());
+                physicalPartition.setNextVersion(version.getVersion() + 1);
+                LOG.info("set partition visible version from {} to {}, " +
+                        "set partition next version from {} to {}, versionInfo: {}, db name: {}, table name: {}",
+                        originVisibleVersion, physicalPartition.getVisibleVersion(), originNextVersion,
+                        physicalPartition.getNextVersion(), version, database.getFullName(), table.getName());
+                removeUnRecoveredPartitions(new UnRecoveredPartition(database.getFullName(),
+                        table.getName(), partition.getName(), physicalPartition.getId(), null));
+            } finally {
+                locker.unLockDatabase(database, LockType.WRITE);
+            }
+        }
+    }
+
+    private void addToUnRecoveredPartitions(UnRecoveredPartition info) {
+        this.lock.writeLock().lock();
+        try {
+            this.unRecoveredPartitions.add(info);
+        } finally {
+            this.lock.writeLock().unlock();
+        }
+    }
+
+    private void removeUnRecoveredPartitions(UnRecoveredPartition info) {
+        this.lock.writeLock().lock();
+        try {
+            this.unRecoveredPartitions.remove(info);
+        } finally {
+            this.lock.writeLock().unlock();
+        }
+    }
+
+    public void fetchProcNodeResult(BaseProcResult result) {
+        this.lock.readLock().lock();
+        try {
+            for (UnRecoveredPartition info : unRecoveredPartitions) {
+                String advise;
+                if (info.getMaxCommonVersion() != -1L) {
+                    advise = String.format("You can set the partition version to %d, using command: " +
```

Comment:
after `admin set max version`, can the tablet recover to ingestable state automatically?